### ISSUE TITLE
 Problems installing locally or brilliantly made with Golem as a package on R

### ISSUE THREAD

**Issue by henrique1008** [ThinkR-open/golem#387]
Good Morning

I had a problem when trying to run a Shiny made with Golem installed locally on my machine.

To reproduce my example, just create a shiny golem normally using the path in Rstudio

File >> New Project >> Package for Shiny App using golem

After creating the project directory, and executing the commands in the file located in `~ / dev / 01_start.R,` I close the project.

Now starts a new session in R, where the application will be installed as a package using

`remotes :: install_local ("path_to_shiny_no_your_computer", force = TRUE)`

In the previous version of Golem, I worked with some shinys like that, but in this version I had problems.

After installation, when trying to run Shiny using

`shiny_name :: run_app ()` I get the following error

```
Listening on http://127.0.0.1:7050
Warning: Error in: No root directory found in D: / path_to_shiny / R or its parent directories. Root criterion: contains a file `DESCRIPTION`
  66: stop
  65: rprojroot_find_root
  64: pkgload :: pkg_path
  60: get_golem_wd
  54: favicon
  51: golem_add_external_resources
  48: ui
```

Is this due to some new configuration I have to do, which was not present in the previous version?

I thank you for your collaboration.

**Comment by VincentGuyader** [comment]
Hi,

I cant reproduce.
can you try to :

1 - create a new golem
2 - dont run anyting from 01_start
3 - close reboot Rsudio
4 - run remotes :: install_local ("path_to_shiny_no_your_computer", force = TRUE)
5 - run shiny_name :: run_app ()

**Comment by ColinFay** [comment]
As far as I see it, this seems to be a favicon issue. 

Can you comment the `favicon()` function inside you `app_ui` ?

**Comment by ColinFay** [comment]
favicon uses `get_golem_wd()` which deports things to `here()`, so it doesn't work if you're not in the R project that contains the favicon

Can reproduce : 

```
cd /tmp/
R -e "golem::create_golem('golex')"
cd golex
R -e "remotes::install_local(upgrade = FALSE, force = TRUE)"
cd ..
R -e "golex::run_app()"
Le chargement a n'ecessit'e le package : shiny

Listening on http://127.0.0.1:7721
Warning: Error in : No root directory found in /private/tmp or its parent directories. Root criterion: contains a file `DESCRIPTION`
  66: stop
  65: rprojroot_find_root
  64: pkgload::pkg_path
  60: get_golem_wd [/Users/colin/Seafile/documents_colin/R/opensource/golem/R/options.R#259]
  54: favicon [/Users/colin/Seafile/documents_colin/R/opensource/golem/R/use_favicon.R#152]
  51: golem_add_external_resources
  48: ui
```

Will work if 

```
cd golex
R -e "golex::run_app()"
```

**Comment by ColinFay** [comment]
quick temporary fix : 

+ Comment the favicon part

+ Add the path to your favicon with `favicon("/www/favicon.ico")` or `favicon("/www/favicon.png"),`

**Comment by ColinFay** [comment]
@henrique1008 can you try with the hot fix I just pushed to dev ?

```
remotes::install_github("Thinkr-open/golem", ref = "dev")
```

**Comment by henrique1008** [comment]
Sorry for the delay, I was in a meeting at the company.

One of my uses for shiny is to install locally on other collaborators' computers, and run via Windows .bat file.

That said, I leave an R script set as follows

`app_name :: run_app ()`, and I create a shortcut on the user's Desktop with a Windows `.bat` batch file that does this.

With the previous answers, I added to the scritp that is run by the Windows` .bat` file

```
setwd ("path_to_shiny_nas_company-shared_folders")
app_name :: run_app ()
```

This worked because the execution file finds the source, similar to the previous solution

```
golex cd
R -e "golex :: run_app ()"
```

Now with this new update that I installed in the "dev" version, installed with

`remotes :: install_github ("Thinkr-open / golem", ref = "dev")`

`Shinys are running normally without the need for setwd ("path")`

Thank you for your availability and readiness.

Golem is the best R package in my opinion, congratulations on the work of the whole team.
